libvncclient/tls_gnutls.c

```diff
@@ -328,7 +328,7 @@ ReadVeNCryptSecurityType(rfbClient* client, uint32_t *result)
         return FALSE;
     }
 
-    if (count>sizeof(tAuth))
+    if (count>(sizeof(tAuth)/sizeof(tAuth[0])))
     {
         rfbClientLog("%d security types are too many; maximum is %d\n", count, sizeof(tAuth));
```

Comment:
We need to fix the maximum size in the message as well:

```suggestion
         rfbClientLog("%d security types are too many; maximum is %d\n", count, (sizeof(tAuth)/sizeof(tAuth[0])));
```